### PR TITLE
#231 | Implement Admin Registration Application Create Endpoint

### PR DESCRIPTION
**Pull request by ibrahimdenizz**
### Description

I added an endpoint to create an admin register application endpoint.

### Related Issue

#231 

### Proposed Changes

Please provide a brief and descriptive title for your pull request, summarizing the changes or features introduced. It
should be in English and clearly indicate the purpose of the pull request. Optionally, you can provide a more detailed
description of what you have done.

- Branches should follow the below `Naming Conventions of Branchs` header in [README.md](https://github.com/afet-yonetim-sistemi/ays-be/edit/main/README.md) File

### Additional Information

If there's any additional information or context that the reviewers should know, please provide it here.

### Checklist

Please check the following before submitting your pull request:

- [X] I have tested my changes locally and they are working as expected.
- [x] The code is formatted according to the project's coding guidelines and style.
- [x] Necessary documentation has been added or existing documentation has been updated.
- [x] Relevant unit tests have been written and included.
- [x] Relevant integration tests have been written and included.
- [x] The code has been reviewed to ensure its quality.
- [x] The code does not contain any issues flagged by SonarLint.
- [x] Default reviewers have been assigned to this pull request.
- [x] Assignees have been added if necessary.
- [x] Labels have been applied if necessary.
- [x] If an issue is opened and this pull request is related to that issue, it should not be associated with any project
  board or milestone. However, if there is no issue, the pull request should be associated with the appropriate project
  board (BE Board, In Review column) and milestone to track the progress.

### Reviewers

The default project maintainers and team members will be automatically assigned as reviewers for this pull request.

### CODE REVIEW

**Comment by agitrubard** [comment]
@ibrahimdenizz conflict mevcut

File: src/main/java/com/ays/admin_user/controller/AdminUserRegisterApplicationController.java [hunk 1]
```diff
@@ -74,4 +79,21 @@ public AysResponse<AdminUserRegisterApplicationResponse> getRegistrationApplicat
...
+     * Requires SUPER_ADMIN authority.
+     *
+     * @param request The request object containing the register application details.
+     * @return A response object containing the created register application.
+     */
+    @PostMapping("/registration-application")
+    @PreAuthorize("hasAnyAuthority('SUPER_ADMIN')")
+    public AysResponse<AdminUserRegisterApplicationCreateResponse> createRegistrationApplication(@RequestBody @Valid AdminUserRegisterApplicationCreateRequest request) {
+        AdminUserRegisterApplication registerApplication = adminUserRegisterApplicationService.createRegistrationApplication(request);
+        System.out.println("Work");
```
Comment (skayikci) [inline src/main/java/com/ays/admin_user/controller/AdminUserRegisterApplicationController.java:93]: remove.

File: src/main/java/com/ays/admin_user/model/dto/request/AdminUserRegisterApplicationCreateRequest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+     * Institution ID for Registering Admin User
+     */
+    @NotBlank
+    private String institutionId;
+
+    /**
+     * Reason for Registering Admin User
+     */
+    @NotBlank
+    @Size(min = 40, max = 512, message = "Reason must be between 40 and 512 characters")
```
Comment (skayikci) [inline src/main/java/com/ays/admin_user/model/dto/request/AdminUserRegisterApplicationCreateRequest.java:26]: message kismina gerek yok.

File: src/main/java/com/ays/common/util/AysRandomUtil.java [hunk 3]
```diff
@@ -53,4 +53,21 @@ public static String generateUUID() {
...
 
+    /**
+     * Generates a random string with the specified length.
+     * The generated string will be a random string with the specified length, consisting of uppercase and lowercase
+     * letters and digits.
+     *
+     * @param length the length of the generated string
+     * @return a random string with the specified length
+     */
+    public static String generateString(int length) {
```
Comment (skayikci) [inline src/main/java/com/ays/common/util/AysRandomUtil.java:64]: sadece test'te kullanılıyorsa test paketine tasiyabilirsin.

File: src/main/java/com/ays/admin_user/controller/AdminUserRegisterApplicationController.java (hunk 1 above)
Comment (ibrahimdenizz) [reply src/main/java/com/ays/admin_user/controller/AdminUserRegisterApplicationController.java]: Kaldırıldı.

File: src/main/java/com/ays/admin_user/model/dto/request/AdminUserRegisterApplicationCreateRequest.java (hunk 2 above)
Comment (ibrahimdenizz) [reply src/main/java/com/ays/admin_user/model/dto/request/AdminUserRegisterApplicationCreateRequest.java]: Kaldırıldı

File: src/main/java/com/ays/common/util/AysRandomUtil.java (hunk 3 above)
Comment (ibrahimdenizz) [reply src/main/java/com/ays/common/util/AysRandomUtil.java]: AysRandomTestUtil class'ına taşındı.

**Review by skayikci — APPROVED** [review]
Lgtm.

**Review by agitrubard — APPROVED** [review]

**Pull request merged**